pkg/config/vault.go

```diff
@@ -51,35 +58,83 @@ func getVaultClientFromConfig(config StepConfig, creds VaultCredentials) (vaultC
 	return &client, nil
 }
 
-func addVaultCredentials(config *StepConfig, client vaultClient, params []StepParameters) {
+func resolveVaultReferences(config *StepConfig, client vaultClient, params []StepParameters) {
 	for _, param := range params {
 		// we don't overwrite secrets that have already been set in any way
 		if _, ok := config.Config[param.Name].(string); ok {
 			continue
 		}
-		ref := param.GetReference("vaultSecret")
-		if ref == nil {
+		resolveVaultSecretReferences(config, client, param)
+		resolveVaultFileReferences(config, client, param)
+	}
+}
+
+func resolveVaultSecretReferences(config *StepConfig, client vaultClient, param StepParameters) {
+	ref := param.GetReference("vaultSecret")
+	if ref == nil {
+		return
+	}
+	var secretValue *string
+	for _, vaultPath := range ref.Paths {
+		// it should be possible to configure the root path were the secret is stored
+		vaultPath, ok := interpolation.ResolveString(vaultPath, config.Config)
+		if !ok {
 			continue
 		}
-		var secretValue *string
-		for _, vaultPath := range ref.Paths {
-			// it should be possible to configure the root path were the secret is stored
-			vaultPath, ok := interpolation.ResolveString(vaultPath, config.Config)
-			if !ok {
-				continue
-			}
 
-			secretValue = lookupPath(client, vaultPath, &param)
-			if secretValue != nil {
-				config.Config[param.Name] = *secretValue
-				log.Entry().Infof("Resolved param '%s' with vault path '%s'", param.Name, vaultPath)
-				break
-			}
+		secretValue = lookupPath(client, vaultPath, &param)
+		if secretValue != nil {
+			config.Config[param.Name] = *secretValue
+			log.Entry().Infof("Resolved param '%s' with vault path '%s'", param.Name, vaultPath)
+			break
 		}
-		if secretValue == nil {
-			log.Entry().Warnf("Could not resolve param '%s' from vault", param.Name)
+	}
```

Comment:
This is also the same as in `resolveVaultFileReferences ` below. Can you extract this so it is reusable for the file case?

Reply:
sure